pkg/quay/quay_test.go

```diff
@@ -425,3 +425,180 @@ func TestQuayClient_DeleteTag(t *testing.T) {
+
+	quayClient := NewQuayClient(client, "authtoken", "https://quay.io/api/v1")
+
+	// Delete existing repository
+	gock.New("https://quay.io/api/v1").
+		MatchHeader("Content-Type", "application/json").
+		MatchHeader("Authorization", "Bearer authtoken").
+		Delete(fmt.Sprintf("repository/%s/%s", org, repo)).
+		Reply(204).
+		JSON(map[string]string{})
+
+	if exists, _ := quayClient.DeleteRepository(org, repo); !exists {
+		t.Error("expected existing repository to be deleted")
+	}
+
+	// Try to delete non-existing repository
+	gock.New("https://quay.io/api/v1").
+		MatchHeader("Content-Type", "application/json").
+		MatchHeader("Authorization", "Bearer authtoken").
+		Delete(fmt.Sprintf("repository/%s/%s", org, repo)).
+		Reply(404).
+		JSON(map[string]string{})
+
+	if exists, _ := quayClient.DeleteRepository(org, repo); exists {
+		t.Error("expected repository to not be deleted since it does not exist")
+	}
+}
+
+func TestQuayClient_ChangeRepositoryVisibility(t *testing.T) {
+	defer gock.Off()
+
+	client := &http.Client{Transport: &http.Transport{}}
+	gock.InterceptClient(client)
+
+	quayClient := NewQuayClient(client, "authtoken", "https://quay.io/api/v1")
+
+	// Change visibility to public
+	gock.New("https://quay.io/api/v1").
+		MatchHeader("Content-Type", "application/json").
+		MatchHeader("Authorization", "Bearer authtoken").
+		BodyString(`{"visibility": "public"}`).
+		Post(fmt.Sprintf("repository/%s/%s/changevisibility", org, repo)).
+		Reply(200).
+		JSON(map[string]string{})
+
+	if err := quayClient.ChangeRepositoryVisibility(org, repo, "public"); err != nil {
+		t.Error("expected change to be successful, err:", err)
+	}
+
+	// Try to change visibility to private without payment
+	gock.New("https://quay.io/api/v1").
+		MatchHeader("Content-Type", "application/json").
+		MatchHeader("Authorization", "Bearer authtoken").
+		BodyString(`{"visibility": "private"}`).
+		Post(fmt.Sprintf("repository/%s/%s/changevisibility", org, repo)).
+		Reply(402).
+		JSON(map[string]string{})
+
+	if err := quayClient.ChangeRepositoryVisibility(org, repo, "private"); err == nil {
+		t.Error("error should have occured")
+	}
+}
+
+func TestQuayClient_GetRobotAccount(t *testing.T) {
+	defer gock.Off()
+
+	client := &http.Client{Transport: &http.Transport{}}
+	gock.InterceptClient(client)
+
+	quayClient := NewQuayClient(client, "authtoken", "https://quay.io/api/v1")
+
+	robot := RobotAccount{
+		Description:  "robot description",
+		Created:      "robot creation",
+		LastAccessed: "robot last access",
+		Token:        "robot token",
+		Name:         "Foo",
+		Message:      "",
+	}
+
+	// Get existing robot account
+	gock.New("https://quay.io/api/v1").
+		MatchHeader("Content-Type", "application/json").
+		MatchHeader("Authorization", "Bearer authtoken").
+		Get(fmt.Sprintf("organization/%s/robots/%s", org, robotName)).
+		Reply(200).
+		JSON(robot)
+
+	if received_robot, err := quayClient.GetRobotAccount(org, robotName); err != nil {
```

Comment:
nit: Is it better to avoid snake case and stick to camel case `received_robot`

Reply:
Thanks, that's what happens when I take a week long break from Go :)